### PR TITLE
Construct crio client once

### PR DESCRIPTION
**Pull request by tedyu**
Signed-off-by: Ted Yu <yuzhihong@gmail.com>

This PR stores crio client in a global variable.
Client() would return this client instance.

### CODE REVIEW

File: container/crio/client.go [hunk 1]
```diff
@@ -28,6 +29,11 @@ const (
 	maxUnixSocketPathSize = len(syscall.RawSockaddrUnix{}.Path)
 )
 
+var (
+	theClient      crioClient
+	clientErr      error
```
Comment (dashpole) [inline container/crio/client.go:34]: clientErr is always nil.  No need to store it.

Comment (tedyu) [reply container/crio/client.go:34]: @dashpole 
Can you take a look at #2460 as well ?
I intend to rebase this after #2460 goes in.

Comment (tedyu) [reply container/crio/client.go:34]: I have rebased the PR.

File: container/crio/client.go [hunk 2]
```diff
@@ -70,16 +77,21 @@ func configureUnixTransport(tr *http.Transport, proto, addr string) error {
...
-	}
-	c := &http.Client{
-		Transport: tr,
-	}
-	return &crioClientImpl{
-		client: c,
-	}, nil
+	crioClientOnce.Do(func() {
+		tr := new(http.Transport)
+		if err := configureUnixTransport(tr, "unix", CrioSocket); err != nil {
```
Comment (dashpole) [inline container/crio/client.go:82]: I think this can just be:
```golang
if clientErr = configureUnixTransport(...; err != nil {
    return
}
```

Comment (tedyu) [reply container/crio/client.go]: @dashpole 
Your comment has been addressed.

**Comment by dashpole** [comment]
cc @runcom @dims

**Comment by dims** [comment]
LGTM logic seems to be same used in docker/client.go

**Pull request merged**